File: flame/rules.py
# -*- coding: utf-8 -*-
import json
import requests

{% macro get(item, url) -%}
def get_{{ item }}():
    api_url = '{{ url }}'
    req = requests.get(api_url)
    assert req.status_code == 200
    return json.loads(req.text)
{% endmacro %}


{% macro post(item, url, obj) -%}
def post_{{ item }}():
    api_url = '{{ url }}'
    headers = {'Content-type': 'application/json'}
    obj = {{ obj }}

    req = requests.post(api_url,
        data=json.dumps(obj), headers=headers)
    assert req.status_code == 200
    return json.loads(req.text)
{% endmacro %}

File: flame/rules.js

{% macro get(item, url) -%}
function get{{ item }}(cb) {
  var apiUrl = '{{ url }}';
  $.get(apiUrl, function(json) {
    var obj = JSON.parse(json);
    cb(obj);
  });
}
{% endmacro %}


{% macro post(item, url, obj) -%}
function post{{ item }}(cb) {
  var apiUrl = '{{ url }}';
  var obj = {{ obj }};
  $.post(apiUrl, obj, function(json) {
    var returnedObj = JSON.parse(json);
    cb(returnedObj);
  });
}
{% endmacro %}
